compiler/src/iree/compiler/InputConversion/MHLO/ConvertCollectiveOps.cpp

```diff
@@ -129,6 +129,36 @@ static IREE::Flow::CollectiveElementTypeAttr getCollectiveElementTypeAttr(
                                                     *collectiveElemType);
 }
 
+template <typename T>
+static LogicalResult checkCollectiveAttrs(T op, PatternRewriter &rewriter) {
+  // Check there is only one group in the replica_groups
+  ShapedType replicaGroupType = op.getReplicaGroups().getType();
+  if (replicaGroupType.getRank() != 2 || replicaGroupType.getDimSize(0) != 1) {
+    return rewriter.notifyMatchFailure(op, "must have a single replica group");
+  }
+
+  int64_t channelHandle =
+      op.getChannelHandle() ? op.getChannelHandleAttr().getHandle() : 0;
+  if (channelHandle <= 0) {
+    if (!op.getUseGlobalDeviceIds()) {
```

Comment:
Nit: lets just check positive case (error case) with comment.

Should this be failing verification of the op? (we normally push invariant checking to the verify method and rely on verified input in passes)

Reply:
Eugene said that there is a default value when an attribute is missing.

Reply:
So channelHandle <=0 && op.getUseGlobalDeviceIds() is a valid combo?

Reply:
Good point, the case is unsupported based on the stablehlo spec. https://github.com/openxla/stablehlo/blob/main/docs/spec.md#parallel-execution

Reply:
I will report this to Eugene.

Reply:
Thank you! This is specced (there are corresponding constraints for all_gather, all_reduce and reduce_scatter in the spec), but we don't have a verifier for this. I've opened a ticket to follow up on this: https://github.com/openxla/stablehlo/issues/1229.